lib/fsm/utils.go

```diff
@@ -41,9 +41,44 @@ func CanRollback(plan *storage.OperationPlan, phaseID string) error {
 		return trace.BadParameter(
 			"phase %q has already been rolled back", phase.ID)
 	}
+	if !latestRollback(plan.Phases, phaseID) {
+		return trace.BadParameter(
+			"rollback subsequent phases before rolling back phase %q", phase.ID)
+	}
 	return nil
 }
 
+// latestRollback returns true if the phase specified by phaseID is next in line
+// to be rolled back.
+func latestRollback(phases []storage.OperationPhase, phaseID string) bool {
+	if len(phases) == 0 {
+		return false
+	}
+
+	// latest keeps track of the latest phase that has not yet been rolled back.
+	var latest storage.OperationPhase
+
+L:
+	for _, phase := range phases {
+		if latestRollback(phase.Phases, phaseID) {
+			return true
+		}
+		switch {
+		// if we run into an unstarted phase, all later phases should also be unstarted
```

Comment:
That's technically not true, a phase can be executed "out of band" using plan execute command so I don't think we can make this assumption.

Reply:
Ah, thanks for pointing that out. Will update.